[Agent Guide] Integration Management (#4172)
DataDog/documentation: content/en/agent/guide/integration-management.md

---
title: Integration Management
kind: guide
---

## Overview

> The Agent comes with a set of bundled official Datadog integrations to allow users to quickly start monitoring their applications. These integrations are also available as single Python packages that can be individually upgraded.

Reviewer: ```suggestion
The Agent comes with a set of bundled official Datadog integrations to allow users to start monitoring their applications quickly. These integrations are available as single Python packages, and you can upgrade them separately.
```

Reviewer: the first sentence is a split infinitive and the second sentence is passive voice that I think could a bit more clear when made active

For Agent v6.8+, the `datadog-agent integration` command allows users to manage the official Datadog integrations that are available for the Agent. It has the following subcommands:

 * [install](#install)
 * [remove](#remove)
 * [show](#show)
 * [freeze](#freeze)

> The usage and documentation of these commands can be printed with `datadog-agent integration --help`.  

Reviewer: ```suggestion
Print the usage and documentation of these commands with `datadog-agent integration --help`.  
```

Reviewer: also passive - making it not passive makes it a bit more concise

For Linux, execute the command as the `dd-agent` user. For Windows, execute the command as an `administrator`.

## Integration commands

> ### Summary workflow

Reviewer 2: ```suggestion
### Workflow
```

1. Check the version of the integration installed in your Agent with the `show` command.
2. Review the changelog of the specific integration on the [integrations-core][1] repository to identify the version you want.
3. Install the integration with the `install` command.
4. Restart your Agent.

> **Note**: When using a configuration management tool, it is recommended to pin the integration to the desired version. When the Agent is upgraded, remove the integration pin. Upgrading the Agent without removing the integration pin can cause the configuration management tool to fail if the version of the integration is not compatible with the new version of the Agent.

Reviewer: ```suggestion
**Note**: When using a configuration management tool, it is recommended to pin the integration to the desired version. When you're ready to upgrade the Agent, remove the pin. Upgrading the Agent without removing the integration pin can cause the configuration management tool to fail if the version of the integration is not compatible with the new version of the Agent.
```

### Install

Use the `datadog-agent integration install` command to install a specific version of an official Datadog integration (available on the [integrations-core repository][1]), provided that it is compatible with the version of the Agent. The command does this verification and exits with a failure in case of incompatibilities.

An integration is compatible and installable if:
 1. The version is newer than the one [shipped with the Agent][2].

>  1. It is compatible with the version of the [datadog_checks_base][3].

Reviewer 2: `It is compatible with the version of the datadog_checks_base`

Which version? Current version available, or the version installed on the host? This sentence isn't very clear.

Author: @dsahni - This was taken from the docs in the `datadog-agent` repo. We would need to check with the engineers.

Reviewer 3: version available in the installed agent
